### PR TITLE
Issue feat PS-2921:If pagination applied, and after that filter gets changed, then pagination should reset to first page

### PR DESCRIPTION
**Pull request by AkshataKatwal16**

### CODE REVIEW

**Review by suvarnakale — APPROVED** [review]

File: src/pages/workspace/content/up-review/index.tsx [hunk 1]
```diff
@@ -103,8 +105,24 @@ const UpForReviewPage = () => {
...
+        const selectedFilters = localSelectedFilters?JSON.parse(localSelectedFilters ): null;
+        const primaryCategory = selectedFilters? selectedFilters : filter.length ? filter : [];
+ 
+        if (prevFilterRef.current !== filter) {
+          offset=0;
+          setPage(0);
+          router.push(
+            {
+              pathname: router.pathname,
+              query: { ...router.query, page: 1 },
```
Comment (itsvick) [inline src/pages/workspace/content/up-review/index.tsx:119]: @AkshataKatwal16 everywhere same code repeated, can u optimize this... and format files

**Review by itsvick — CHANGES_REQUESTED** [review]

File: src/pages/workspace/content/up-review/index.tsx (hunk 1 above)
Comment (AkshataKatwal16) [reply src/pages/workspace/content/up-review/index.tsx:119]: @itsvick should i write common function for that?
```
if (prevFilterRef.current !== filter) {
          offset=0;
          setPage(0);
          router.push(
            {
              pathname: router.pathname,
              query: { ...router.query, page: 1 }, 
            },
            undefined,
            { shallow: true } 
          );
          prevFilterRef.current = filter;
        }
```

Comment (AkshataKatwal16) [reply src/pages/workspace/content/up-review/index.tsx]: @itsvick moved that router logic to common component  searchbox

File: .env [hunk 2]
```diff
@@ -1,12 +1,7 @@
...
...EtNWJmZC00ZjMwLTljMzktNGViODc3N2EyNzEyIiwiaXNzIjoiaHR0cHM6Ly9rZXljbG9hay5wcmF0aGFtZGlnaXRhbC5vcmcvYXV0aC9yZWFsbXMvcHJhdGhhbSIsImF1ZCI6ImFjY291bnQiLCJzdWIiOiIzZTBiZGE2Yi1jNTA2LTQ2NmYtODg3NS1lMzdkZTUyYTBlOGYiLCJ0eXAiOiJCZWFyZXIiLCJhenAiOiJwcmF0aGFtIiwic2Vzc2lvbl9zdGF0ZSI6IjRkYWI2YWRhLWZmYWItNGZkZS05N2UwLWQwYTZkNTAyMDliOSIsImFjciI6IjEiLCJhbGxvd2VkLW9yaWdpbnMiOlsiLyoiXSwicmVhbG1fYWNjZXNzIjp7InJvbGVzIjpbIm9mZmxpbmVfYWNjZXNzIiwidW1hX2F1dGhvcml6YXRpb24iLCJkZWZhdWx0LXJvbGVzLXByYXRoYW0iXX0sInJlc291cmNlX2FjY2VzcyI6eyJhY2NvdW50Ijp7InJvbGVzIjpbIm1hbmFnZS1hY2NvdW50IiwibWFuYWdlLWFjY291bnQtbGlua3MiLCJ2aWV3LXByb2ZpbGUiXX19LCJzY29wZSI6ImVtYWlsIHByb2ZpbGUiLCJzaWQiOiI0ZGFiNmFkYS1mZmFiLTRmZGUtOTdlMC1kMGE2ZDUwMjA5YjkiLCJlbWFpbF92ZXJpZmllZCI6ZmFsc2UsIm5hbWUiOiJSYWh1bCIsInByZWZlcnJlZF91c2VybmFtZSI6InJhaHVsX3Rla2RpIiwiZ2l2ZW5fbmFtZSI6IlJhaHVsIiwiZmFtaWx5X25hbWUiOiIifQ.TKrgbwBQe8NnpzqxgxtltzD_gFXs9XKDvLjqmAUZoAZAOtxd66vWGuE8VJouVrFMKbrLy-HeX8cf4Uh39wrPJzYcK5vayi2J0fCmMK84XechQIDOT7oIQ6rbKB7mpeV8tM3kBluoHvjLt0-Kh71JY0kv_WQ8eZCpC2VygM9dQcl79UUkSWr4KD9T4T_sdhCITYazPEntbix1TknFyavZoCnvqSkPg6ARxStTbNDnf7HmQWAqtg1XpVLB7CUT2IoBJjGwB2uAILZ4EglkhZD9I4N8PhlPRTiyjWX_E5w-tAeGZOdHO5bW8wGyC_JyFPbk1Z_1sqwIvGECyYLgDqBr7w"
```
Comment (itsvick) [inline .env:2]: @AkshataKatwal16 dont push env file

File: src/pages/workspace/content/allContents/index.tsx [hunk 3]
```diff
@@ -129,31 +152,41 @@ const AllContentsPage = () => {
...
             "FlagReview",
           ];
-        }
-        else if (statusBy === "Live") {
+        } else if (statusBy === "Live") {
           status = ["Live"];
-        }
-        else if (statusBy === "Review") {
+        } else if (statusBy === "Review") {
           status = ["Review"];
```
Comment (itsvick) [inline src/pages/workspace/content/allContents/index.tsx:158]: @AkshataKatwal16 use switch case here

File: src/pages/workspace/content/discover-contents/index.tsx [hunk 4]
```diff
@@ -121,40 +144,51 @@ const ContentsPage = () => {
...
+        const contentType = "discover-contents";
+        let response;
+        if (state !== "All") {
+          response = await getContent(
+            status,
+            query,
+            LIMIT,
+            offset,
+            primaryCategory,
+            sort_by,
```
Comment (itsvick) [inline src/pages/workspace/content/discover-contents/index.tsx:176]: @AkshataKatwal16 optimize below code by just 
```const response = await getContent(
  status,
  query,
  LIMIT,
  offset,
  primaryCategory,
  sort_by,
  contentType,
  state !== "All" ? state : undefined
);
```

File: src/pages/workspace/content/submitted/index.tsx [hunk 5]
```diff
@@ -87,15 +100,28 @@ const SubmittedForReviewPage = () => {
...
       try {
         setLoading(true);
         const query = debouncedSearchTerm || "";
-        const offset =debouncedSearchTerm!==""? 0 : page * LIMIT;
-        const primaryCategory = filter.length ? filter : [];
+        let offset = debouncedSearchTerm !== "" ? 0 : page * LIMIT;
+        const localSelectedFilters = localStorage.getItem("selectedFilters");
+        const selectedFilters = localSelectedFilters
+          ? JSON.parse(localSelectedFilters)
+          : null;
```
Comment (itsvick) [inline src/pages/workspace/content/submitted/index.tsx:112]: @AkshataKatwal16 code is repeated everywhere... try to optimize this

File: src/pages/workspace/content/allContents/index.tsx (hunk 3 above)
Comment (AkshataKatwal16) [reply src/pages/workspace/content/allContents/index.tsx]: done

File: src/pages/workspace/content/discover-contents/index.tsx (hunk 4 above)
Comment (AkshataKatwal16) [reply src/pages/workspace/content/discover-contents/index.tsx]: done

File: .env (hunk 2 above)
Comment (AkshataKatwal16) [reply .env]: reverted back env

File: src/pages/workspace/content/submitted/index.tsx (hunk 5 above)
Comment (AkshataKatwal16) [reply src/pages/workspace/content/submitted/index.tsx]: set by route query now

**Pull request merged**